### PR TITLE
Validate nonconventional numeric metadata values are numeric (SCP-3876)

### PR DESCRIPTION
**Pull request by jlchang**
Currently, nonconventional numeric metadata are not checked to ensure submitted values are numeric. Add a check to detect if nonconventional metadata of TYPE numeric have invalid values.

To test, set your local instance "Ingest Pipeline Docker Image" configuration to use
`gcr.io/broad-singlecellportal-staging/scp-ingest-jlc_process_nonconventional_numeric:66b09c6`  

and upload metadata file:
`https://raw.githubusercontent.com/broadinstitute/scp-ingest-pipeline/jlc_process_nonconventional_numeric/tests/data/annotation/metadata/convention/invalid_nonconventional_numeric_v2.2.0.txt`

Look for the following error message the ingest failure email:

> *** type error list:
> percent_mt: supplied value 07.juil is not numeric [ Error count: 1 ]

[Context](https://docs.google.com/presentation/d/1AXkXUQrH_vHsJMJ_Zo1xYU5XM9RXkYbpme2oK2ampsE/edit#slide=id.g10453e60d48_0_0) for why ingest will generate variable error messages for this validation

This PR supports SCP-3876 by addressing [Zendesk 203931](https://broadinstitute.zendesk.com/agent/tickets/203931) but may not address the [originating Sentry issue](https://sentry.io/organizations/broad-institute/issues/2356928042/?project=1424198) (lacked input files to replicate issue).

### CODE REVIEW

File: ingest/validation/validate_metadata.py [hunk 1]
```diff
@@ -915,8 +915,24 @@ def process_metadata_row(metadata, convention, line):
     row_info = dict(itertools.zip_longest(metadata_names, line))
     processed_row = {}
     for k, v in row_info.items():
-        # for metadata not in convention, no need to process
+        # for metadata not in convention, check numeric metadata for valid values
         if k not in convention["properties"].keys():
+            type_index = metadata.headers.index(k)
+            k_type = metadata.annot_types[type_index]
+            if k_type == "numeric":
+                k_numeric = type(v) == int or type(v) == float
```
Comment (eweitz) [inline ingest/validation/validate_metadata.py:923]: Faster, and more concise:

```suggestion
                k_numeric = type(v) in [int, float]
```

For reference:
```
>>> foo = 1
>>> bar = 1.2
>>> baz = "hi"
>>> type (foo) in [int, float]
True
>>> type (bar) in [int, float]
True
>>> type (baz) in [int, float]
False
```

File: ingest/validation/validate_metadata.py [hunk 2]
```diff
@@ -915,8 +915,24 @@ def process_metadata_row(metadata, convention, line):
...
+                if not k_numeric:
+                    # pandas coercion will stringify numbers
+                    try:
+                        float(v)
+                        msg = f"{k}: one or more values are not numeric"
```
Comment (eweitz) [inline ingest/validation/validate_metadata.py:928]: Could you clarify, at least in a code comment if not also the `msg`, a specific case or two that trigger this?

```suggestion
                        msg = f"{k}: one or more values are not numeric"
```

**Review by eweitz — APPROVED** [review]
Code looks good!  I suggest a trivial optimization / readability refinement, and a bit of clarification in one of the code paths.

File: ingest/validation/validate_metadata.py (hunk 1 above)
Comment (jlchang) [reply ingest/validation/validate_metadata.py:923]: Thanks! That is much more readable :)

File: ingest/validation/validate_metadata.py (hunk 2 above)
Comment (jlchang) [reply ingest/validation/validate_metadata.py]: Done! c91ca9b added an example in the comments and, hopefully, this updated message is more precise:
`one or more values in data column are not numeric`

**Review by bistline — APPROVED** [review]

**Pull request merged**